addon/services/file-manager.js

```diff
@@ -141,25 +244,65 @@ export default Ember.Service.extend({
                 headers: {
                     Authorization: `Bearer ${accessToken}`
                 },
-                success: (data) => resolve(data),
-                error: (_, __, error) => reject(error)
+                error: (_, __, error) => reject(error),
+                success: (data) => resolve(data)
             });
         });
     },
 
-    _pushToStore(data, id, parentFolder) {
-        // Hack the Waterbutler entity to look like the file Ember expects
-        data.data.id = id;
-        data.data.type = 'file';
-        let attr = data.data.attributes;
-        attr.materializedPath = attr.materialized;
-        attr.dateModified = attr.modified;
+    /**
+     * Get an updated `file` model from the OSF API.
+     *
+     * @method _reloadModel
+     * @private
+     * @param {Object} model Ember model to reload.
+     * @param {Function} isFresh Function that is called with the reloaded
+     * model from the OSF API. If the model is actually still stale, should
+     * return `false`. Otherwise, return the fresh model.
+     * @return {Promise} Promise that resolves to an updated `file` model or
+     * rejects with an error message.
+     */
+    _reloadModel(model, isFresh) {
+        const maxTries = 32;
```

Comment:
I found that reloading the model immediately after the waterbutler request returns would usually-to-always yield the old, pre-waterbutler-action file info, but trying again after a little wait would work as expected. Instead of asking someone about this, I accepted and worked around it with the `isFresh` function to check whether it needs to try reloading again.

This is, of course, extremely hacky. I chose powers of two because it made the hackiness ever-so-slightly more palatable. If this reflects a solvable server-side problem, that would simplify file-manager nicely. I'll verify that I wasn't hallucinating and can reproduce the behavior, then annoy Fitz.